dipy/utils/arrfuncs.py

```diff
@@ -28,3 +32,97 @@ def as_native_array(arr):
         return arr
     return arr.byteswap().newbyteorder()
 
+
+def pinv(a, rcond=1e-15):
+    """Vectorized version of `numpy.linalg.pinv`
+
+    If numpy version is less than 1.8, it falls back to iterating over
+    `np.linalg.pinv` since there isn't a vectorized version of `np.linalg.svd`
+    available.
+
+    Parameters
+    ----------
+    a : (..., M, N) array_like
```

Comment:
Just a stylistic issue here. In Dipy we first write array_like and then the shape. Please change everywhere.